department-of-veterans-affairs/vets-api
# Bdex/54508: intent to file get abstraction

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.


## Summary

- abstracted calls to the EVSS Intent to File endpoints
- Benefits Disability Experience Team 
- FEATURE_TOGGLE_INTENT_TO_FILE = 'disability_compensation_lighthouse_intent_to_file_provider'

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#54508
- https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/54508
- [*Link to previous change of the code/bug (if applicable)*](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/56799)
- [*Link to epic if not included in ticket*](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/53419)

## Testing done

- Build and start vets-api
- yarn watch vets-website
- Navigate to localhost
- Click 'sign in', in the modal select ID.me (use credentials [vets.gov](http://vets.gov/)[.user+228@gmail.com](mailto:vets.gov.user+228@gmail.com), <pw in internal repo> note: 228 has the information needed to enter the 526 form)
- Navigate straight to the form wizard: http://localhost:3001/disability/file-disability-claim-form-21-526ez/start because navigation might not load (this is because of content-build)
- "Are you on active duty right now?" > "No" > "I'm filing a new claim", or skip the form by navigating to the "If you know X form is right, click here"
- "Start the disability compensation form"
- Verify via the dev tools network tab or with a breakpoint that the first intent to file GET is called
- Verify that a POST is called as well
- Verify the structure of the intenttofile object
- Repeat on master and verify the behavior matches

## Screenshots
![image](https://user-images.githubusercontent.com/19178435/232622389-0bba359a-14cd-40b2-a3e3-65baa57c4627.png)

## What areas of the site does it impact?
Form 21-526EZ

## Acceptance criteria

- [x] Form 526: no changes to intent to file functionality
- [x] Maintain test coverage?


## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
